experiment-newton.ipynb

```diff
@@ -0,0 +1,181 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #7.        grad = pt.linalg.solve(jac, f_x, assume_a="sym")
</pre><p>You're assuming jacobian is symmetrical, but that shouldn't be true in general right?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pytensor/pull/944/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Reply:
Oh, you are right. It is in the case of minimization, but might not be for different root finding problems. We should add an option for that.

Reply:
Ah, good catch! The jacobian is indeed generally asymmetrical for n>1.

At first I thought you were talking about the Hessian. That is symmetrical but with a really weird caveat for the case when things are twice differentiable but the second derivatives aren't continuous. (In these sorts of cases you can get some really [weird stuff](https://tensorial.com/math/funnyfns/whitney/index.html) like a fractal trail that's flat at every point but ascends.)